Gems/ROS2/Code/Source/Spawner/ROS2SpawnerComponent.cpp

```diff
@@ -110,12 +111,22 @@ namespace ROS2
     void ROS2SpawnerComponent::SpawnEntity(
         const SpawnEntityServiceHandle service_handle, const std::shared_ptr<rmw_request_id_t> header, const SpawnEntityRequest request)
     {
+        AZStd::string referenceFrame(request->reference_frame.c_str());
+        const bool isWGS{ referenceFrame == "wgs84" && m_controller.GetSupportWGS() };
+
+        SpawnEntityResponse response;
+
+        if (isWGS && !GeoreferenceRequestsBus::HasHandlers())
+        {
+            response.success = false;
+            response.status_message = "Level is not geographically positioned. Action aborted.";
+            return;
```

Comment:
Don't you need to send back the response before terminating?
```cpp
service_handle->send_response(*header, response);
```

Reply:
Good point!